src/ParagraphsPastePluginBase.php

```diff
@@ -50,8 +73,78 @@ public static function create(ContainerInterface $container, array $configuratio
       $configuration,
       $plugin_id,
       $plugin_definition,
-      $container->get('entity_type.manager')
+      $container->get('entity_type.manager'),
+      $container->get('entity_field.manager'),
+      $container->get('plugin.manager.entity_reference_selection'),
     );
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function build($input) {
+    $property_path = explode('.', $this->pluginDefinition['property_path']);
+
+    $target_entity_type = array_shift($property_path);
+    $target_bundle = array_shift($property_path);
+
+    $entity_type = $this->entityTypeManager->getDefinition($target_entity_type);
+
+    $paragraph_entity = $this->entityTypeManager->getStorage($target_entity_type)
+      ->create([
+        $entity_type->getKey('bundle') => $target_bundle,
+      ]);
+
+    $this->setValue($property_path, $paragraph_entity, $input);
+
+    return $paragraph_entity;
+  }
+
+  /**
+   * Sets value to a property path.
+   *
+   * @param array $property_path
+   *   Property path.
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   Entity to set the value on.
+   * @param mixed $value
+   *   The value to set.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  protected function setValue(array $property_path, EntityInterface $entity, $value) {
+
+    $fields = $this->entityFieldManager->getFieldDefinitions($entity->getEntityTypeId(), $entity->bundle());
+
+    $path = array_shift($property_path);
+
+    list($fieldName, $bundle) = array_pad(explode(':', $path), 2, NULL);
+
+    if (empty($fields[$fieldName])) {
+      return;
+    }
+
+    $fieldConfig = $fields[$fieldName];
+
+    // If bundle is defined, it's an ER field.
+    if ($bundle) {
+
+      $target_type = $fieldConfig->getFieldStorageDefinition()
+        ->getSetting('target_type');
+
+      /** @var \Drupal\Core\Entity\EntityInterface $newEntity */
+      $newEntity = $this->entityReferenceSelectionManager->getSelectionHandler($fieldConfig)
+        ->createNewEntity($target_type, $bundle, 'Created by paragraphs paste', 0);
+
+      $entity->{$fieldName}[] = $newEntity;
+      if (!empty($property_path)) {
+        $this->setValue($property_path, $newEntity, $value);
+      }
+    }
+    else {
+      $entity->{$fieldName} = $value;
+    }
+    $entity->save();
```

Comment:
Saving the entity at this point will lead to stale entities in database with no parent (host) entity in paragraphs case. Pass it through to the build function to only have it in the form.